### DISCUSSION TITLE
Possibility of a timeout configuration

### DISCUSSION THREAD

**Discussion by ckipp01** [mfussenegger/nvim-dap#314] [Q&A] [answered]
I noticed here, that there is a default of 2 seconds before a warning is shown to the user: https://github.com/mfussenegger/nvim-dap/blob/208f17456b26d82d4970c2fe46845ac14812280f/lua/dap/session.lua#L816

I'm curious if you'd be open to having a different default being passed in for this, and if so what would you preference be on how to pass it in? When using [nvim-metals](https://github.com/scalameta/nvim-metals) I see this message quite a bit, so it'd be nice to be able to bump this a bit.

**Comment by mfussenegger** [comment, marked as answer]
Sounds good to me. Would you want to configure this as part of the adapter, or have it as user configuration?


I'm also fine with bumpin the built-in value a bit. But I wouldn't go too high as the "nothing happens" experience if something is not setup right (and the adapter doesn't return an error, but does just nothing) isn't ideal either. But 3 or even 4 seconds might still be okay?

**Reply by ckipp01** [reply]
> Sounds good to me. Would you want to configure this as part of the adapter, or have it as user configuration?

Good question. At first my mind went towards being part of an adapter since it can be pretty language specific, but looking at what's available in the current adapter configurations, there really isn't other configuration settings like this there, so maybe it would be better as a user configuration. Did you have a suggestion or opinion on where this would fit better?

> But 3 or even 4 seconds might still be okay?

Yea I think that _should_ be good.

**Comment by ckipp01** [comment]
I wanted to revisit this if that's alright. I've been using this more lately on some projects that are larger than the ones I've tested with in the past. Almost every time I run a test I hit on this, and I've gotten feedback that other users in Scala have as well. I think just because of the nature of Scala and having to ensure everything is compiled first, then some extra stuff on the server side to grab the tests potentially in the file, and then finally return the results, the 4 seconds isn't enough.

Looking at it, what I was thinking would be nice was to be able to just update this in my `enrich_config` here:

https://github.com/scalameta/nvim-metals/blob/86dadf8c68b93037d7e1dc4b47f731e52bda0514/lua/metals/setup.lua#L146-L151

I could default it to a bit higher, and then a user could also add it to their config if they wanted to change it. Then this would just be passed in as part of the config to the initialize:

https://github.com/mfussenegger/nvim-dap/blob/b17d513beee21b113cb99c501b8e8325f5dccb61/lua/dap/session.lua#L817

If present use it, or else default to the current. Does this sound alright to you? If so I'll shoot in a pr.
